pkg/netpol/eval/check.go

```diff
@@ -436,3 +351,271 @@ func GetPeerExposedTCPConnections(peer Peer) *common.ConnectionSet {
 		return nil
 	}
 }
+
+// allAllowedXgressConnections: returns the allowed connections from srcPeer to dstPeer on the
+// given direction (ingress/egress)
+// allowed conns are computed by walking through all the available resources of k8s network policy api:
+// admin-network-policies, network-policies and baseline-admin-network-policies;
+// considering the precedence of each policy
+func (pe *PolicyEngine) allAllowedXgressConnections(src, dst k8s.Peer, isIngress bool) (allowedConns *common.ConnectionSet, err error) {
+	// first get allowed xgress conn between the src and dst from the ANPs
+	// note that:
+	// - anpConns may contain allowed, denied or/and passed connections
+	// - anpCaptured is true iff there is at least one rule in the input ANPs that captures both src and dst;
+	// because anp rules are read as is and don't contain any implicit isolation effects for the Pods selected by the AdminNetworkPolicy.
+	anpConns, anpCaptured, err := pe.getAllAllowedXgressConnectionsFromANPs(src, dst, isIngress)
+	if err != nil {
+		return nil, err
+	}
+	// optimization: if all the conns between src and dst were determined by the ANPs : return the allowed conns
+	if anpCaptured && anpConns.DeterminesAllConns() {
+		return anpConns.AllowedConns, nil
+	}
+	// second get the allowed xgress conns between the src and dst from the netpols
+	// note that :
+	// - npConns contains only allowed connections
+	// - npCaptured is true iff there are policies selecting either src or dst based on the checked direction (ingress/ egress)
+	npConns, npCaptured, err := pe.getAllAllowedXgressConnsFromNetpols(src, dst, isIngress)
+	if err != nil {
+		return nil, err
+	}
+
+	// compute the allowed connections on the given direction considering the which policies captured the xgress connection
+	// and precedence of each policy type:
+	switch npCaptured {
```

Comment:
the `switch` can be rewritten for better clarity:
```
switch{
  case npCaptured && !anpCaptured : 
   ...
  case npCaptured && anpCaptured : 
   ...
  case !npCaptured:
   ...
}
```

Reply:
done